src/main/java/org/folio/service/file/SplitFileProcessingService.java

```diff
@@ -0,0 +1,133 @@
+package org.folio.service.file;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.ext.web.handler.HttpException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.HttpStatus;
+import org.folio.dao.DataImportQueueItemDao;
+import org.folio.rest.client.ChangeManagerClient;
+import org.folio.rest.impl.util.BufferMapper;
+import org.folio.rest.jaxrs.model.DataImportQueueItem;
+import org.folio.rest.jaxrs.model.File;
+import org.folio.rest.jaxrs.model.InitJobExecutionsRqDto;
+import org.folio.rest.jaxrs.model.JobExecution;
+import org.folio.rest.jaxrs.model.JobExecutionDto;
+import org.folio.rest.jaxrs.model.JobExecutionDtoCollection;
+import org.folio.rest.jaxrs.model.UploadDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service containing methods to manage the lifecycle and initiate processing of
+ * split files.
+ */
+@Service
+public class SplitFileProcessingService {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private DataImportQueueItemDao queueItemDao;
+
+  @Autowired
+  public SplitFileProcessingService(DataImportQueueItemDao queueItemDao) {
+    this.queueItemDao = queueItemDao;
+  }
+
+  /**
+   * Registers split parts as Job Executions in mod-source-record-manager
+   * and adds each part to the DI queue.
+   *
+   * @return a {@link CompositeFuture} of {@link JobExecutionDto}
+   */
+  public CompositeFuture registerSplitFiles(
```

Comment:
could you please add a note to the comment for the method parameters to describe what `keys` means. Does it mean a path/location of the part of the original split file?

Reply:
Yes; key is basically a file name for S3

Reply:
got it thanks! Propose to add a note in the method's comment with a brief description of the method's parameters using @param Javadoc tag.